### ISSUE TITLE
Ability to nest components

### ISSUE THREAD

**Issue by VojtechPetru** [wrabit/django-cotton#94]
Hello 👋 

Thanks for an amazing library, I've been trying it out and its very exciting!

I've got one question - is it possible to use a component inside of another component's definition?

**Minimalistic example:**
```html
<!-- cotton/cards/card.html -->

<div class="bg-white shadow rounded border p-4">
    <p>{{ slot }}</p>
</div>
```

```html
<!-- cotton/cards/special_card.html -->

<c-cards.card style="background: red;">
    <p>{{ slot }}</p>
</c-cards.card>
```

If used in a view:
```html
<!-- view.html -->

<c-cards.special-card>
    Very special card
</c-cards.special-card>
```

The HTML contains the unrendered component tags (inner component is not rendered)
```html
<c-cards.card style="background: red">
    <p>Very special card</p>
</c-cards.card>
```

Expected result:
```html
<div class="bg-white shadow rounded border p-4" style="background: red">
    <p>Very special card</p>
</div>
```

Am I doing something wrong, or is this not supported?

**Comment by wrabit** [comment]
@VojtechPetru your minimal example works for me as expected. The result I get (noting that you have double `p` tags) is:

```html
<div class="bg-white shadow rounded border p-4">
    <p><p>Very special card</p></p>
</div>
```

What version of cotton are you using?
Are you getting the same results from your minimal example?

**Comment by VojtechPetru** [comment]
Hello @wrabit 👋 thanks for such a quick response! I'm already AFK, so I'll re-try and share more info later. Glad to hear that its supposed to work and the problem is likely on my end. Again, amazing library ❤️

**Comment by VojtechPetru** [comment]
Hello @wrabit 👋 I just tested it on a fresh project and it works as expected (except that I forgot to add `{{ attrs }}` to `cotton/cards/card.html` in the example). Must be some other dependency or settings in the old project thats causing the issue - I'll try to figure it out and open a new issue if I do. Closing this issue, thanks!

_On a side note - is this project open to contributions? I've been exploring the source code a bit and would happily add type annotations to make it more accessible, if you'd find it helpful._
